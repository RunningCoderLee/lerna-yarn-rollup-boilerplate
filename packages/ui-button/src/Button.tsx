import { PropsWithChildren, useCallback } from 'react'
import clsx from 'clsx'
import { sayHelloTo } from '@lyrb/ui-utils'
import './style.css'

interface Props {
  type: 'primary' | 'default'
}

const Button = (props: PropsWithChildren<Props>) => {
  const { type } = props

  const classNames = clsx('btn', {
    'btn-primary': type === 'primary',
  })

  const handleClick = useCallback(() => {
    sayHelloTo('John')
  }, [])

  return (
    <button className={classNames} type="button" onClick={handleClick}>
      {props.children}
    </button>
  )
}

export default Button
