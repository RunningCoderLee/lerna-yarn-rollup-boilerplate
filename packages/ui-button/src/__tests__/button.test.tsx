import renderer from 'react-test-renderer'
import Button from '../Button'

describe('test button component', () => {
  it('render correctly', () => {
    const component = renderer.create(<Button type="primary">123</Button>)

    expect(component).toMatchSnapshot()
  })
})
