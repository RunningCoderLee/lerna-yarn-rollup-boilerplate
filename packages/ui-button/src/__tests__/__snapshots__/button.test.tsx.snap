// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`test button component render correctly 1`] = `
<button
  className="btn btn-primary"
  onClick={[Function]}
  type="button"
>
  123
</button>
`;
